src/WalkingModule/include/WalkingControllers/WalkingModule/Module.h

```diff
@@ -66,6 +70,68 @@ namespace WalkingControllers
+        double m_impactTimeAdjusted;/**< Adjusted absolute time of the impact */
+        iDynTree::Vector2 m_zmpNominal;/**< Nominal 2D zmp position */
+        iDynTree::Vector2 m_zmpAdjusted;/**< Absolute 2D zmp position */
+        int m_indexPush;/**< Number of control cycle that step adjustment is active */
+
+        iDynTree::Vector2 m_dcmEstimatedI; /**< The estimated DCM. */
+
+        std::vector<std::shared_ptr<GeneralSupportTrajectory>> m_DCMSubTrajectories;/**< The different trajectories that are output of DCM motion planing. */
+
+        iDynTree::Transform m_adaptedFootLeftTransform;/**< The adapted transform of left foot. */
+        iDynTree::Transform m_adaptedFootRightTransform;/**< The adapted transform of right foot. */
+        iDynTree::Twist m_adaptedFootRightTwist;/**< The adapted twist of right foot. */
+        iDynTree::Twist m_adaptedFootLeftTwist;/**< The adapted twist of left foot. */
+        iDynTree::SpatialAcc m_adaptedFootLeftAcceleration;/**< The adapted acceleration of the left foot. */
+        iDynTree::SpatialAcc m_adaptedFootRightAcceleration;/**< The adapted acceleration of the right foot. */
+
+        iDynTree::Transform m_currentFootLeftTransform;/**< The current adapted transform of the left foot that has been found in previous control cycle. */
+        iDynTree::Transform m_currentFootRightTransform;/**< The current adapted transform of the right foot that has been found in previous control cycle. */
+        iDynTree::Twist m_currentFootLeftTwist;/**< The current adapted twist of  the left foot that has been found in previous control cycle. */
+        iDynTree::Twist m_currentFootRightTwist;/**< The current adapted twist of  the right foot that has been found in previous control cycle. */
+        iDynTree::SpatialAcc m_currentFootLeftAcceleration; /**< The current adapted acceleration of  the left foot that has been found in previous control cycle. */
+        iDynTree::SpatialAcc m_currentFootRightAcceleration;/**< The current adapted acceleratiom of  the right foot that has been found in previous control cycle. */
+
+        std::shared_ptr<FootPrint> m_jleftFootprints; /**< The left foot prints */
+        std::shared_ptr<FootPrint> m_jRightFootprints;/**< The right foot prints */
+
+        iDynTree::Vector2 m_zmpToCenterOfFootPositionLeft;
+        iDynTree::Vector2 m_zmpToCenterOfFootPositionRight;
+
+        StepList m_jLeftstepList; /**< The list of left foot steps */
+        StepList m_jRightstepList;/**< The list of right foot steps */
+
+        StepAdapterOutput m_outputStepAdaptation;/**< The structure of outputs of step adaptation */
+        StepAdapterInput m_inputStepAdaptation;/**< The structure of inputs of step adaptation */
+
+        double m_isPushActive;/**< Is push recovery active? */
+        double m_isRollActive;/**< Is the threshold of roll angles of arm active? */
+        double m_isPitchActive;/**< Is the threshold of pitch angles of arm active? */
```

Comment:
I'm sorry but I really don't like this. Using this kind threshold for detecting the push is something strongly related to the robot we are using. The structure of the `walking-controller` allows using the very same architecture with different kinds of humanoid as we did in the past with @lrapetti and @prajval10.

Reply:
Actually as we discussed T2T we will solve it generally after some other modification inside walking module.